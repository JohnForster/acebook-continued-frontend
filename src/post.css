.post {
  background-color: white;
  margin: 20px auto 20px auto;
  padding: 15px 15px 10px 15px;
  width: 660px;
}

.post_body {
  padding: 13px 0 10px 0;
}

.post_header {
  border-bottom: 1px solid #D1D4D7;
  line-height: 22px;
  padding: 0 0 10px 0;
}

#post_username {
  color: #385899;
  font-weight: bold;
}

#post_time {
  color: #ABAEAF;
  font-size: 13px;
}