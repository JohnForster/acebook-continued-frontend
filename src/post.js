import React from "react";
import {Comment} from './comment';
import css from './post.css';

export class Post extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      message: props.message,
      time: props.time,
      id: props.id,
      comments:[],
    }
  }

  render() {
    const comments = this.state.comments;
    let message = this.state.message;
    let time = this.state.time
    return (
      <div className="post">
        <div className="post_header">
          <p id="post_username">Username</p>
          <p id="post_time">{time}</p>
        </div>
        <div className="post_body">
          {message}
        </div>
        <div className="post_footer">
          {commentsList(comments)}
        </div>
      </div>
    )
  }

  componentDidMount() {
    let id = this.state.id
    fetch('http://localhost:3000/comments?post_id=' + id)
    .then(response => response.json())
    .then(data =>
      this.setState({comments: data})
    )
  }
}

function commentsList(comments) {
  return comments.map(function(comment) {
    return <Comment
      body={comment.body}
      time={comment.created_at}
      user={comment.user}
      />
    })
}