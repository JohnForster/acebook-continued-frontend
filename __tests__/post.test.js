import React from 'react';
import expect from 'expect';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import {Post} from '../src/post.js';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('Post', () => {
  it('Post renders the message given', () => {
    const component = shallow(<Post message='Hello' time='Today'/>);
    expect(component).toMatchSnapshot();
  });
});

// const component = renderer.create(<Post message='Hello' time='Today' />);
