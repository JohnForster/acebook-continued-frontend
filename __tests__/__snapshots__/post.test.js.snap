// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Post renders the message given 1`] = `
<div
  className="post"
>
  Hello
   
  <br />
   (posted at 
  Today
  )
</div>
`;
